### PR TITLE
Fcc 119 importer status endpoint

### PR DESCRIPTION
**Pull request by kpennels**
**Updated Files**
Analytics/app.py -  imported the Import Status resource and exposed the /importer_status endpoint
Analytics/importers/config.yml - added missing keys to config file 
Analytics/importers/scheduler_new.py - added a SQLAlchemy jobstore which persists the importer jobs

**New Files**
Analytics/models/importer_status.py - the model class for the Importer status DB table and the corresponding class methods 
Analytics/models/test_import_status.py -  tests for the /importer_status endpoint

### CODE REVIEW

File: Analytics/importers/config_decorator.py [hunk 1]
```diff
@@ -56,8 +54,10 @@ def get_config(self, *args: str):
                     with open(self.config_file, 'r') as ymlfile:
                         self.config = yaml.load(ymlfile)
                 except FileNotFoundError as e:
-                    importer_status.status = Status.failure(__class__.__name__, e.__str__(), traceback.format_exc())
-                    exit()
+                    importer_status.status = Status.failure(
+                        self.__class__.__name__, e.__str__(),
+                        traceback.format_exc())
+                    return
```
Comment (j0rdanl0urens) [inline Analytics/importers/config_decorator.py]: Shouldn't this be `return None`

File: Analytics/importers/scheduler_new.py [hunk 2]
```diff
@@ -7,105 +7,141 @@
...
-        self.scheduler.start()
-
-    def get_apis(self) -> list:
+    @status_tracker.changed.register
+    def status_has_changed(self, status: ImporterStatuses):
+        """
+        Receive the status of an importer and persist it to the Importer
+        Status table
+        :param status: Status object identifying the
+        :return:
```
Comment (j0rdanl0urens) [inline Analytics/importers/scheduler_new.py]: Remove since not used

File: Analytics/importers/scheduler_new.py [hunk 3]
```diff
@@ -7,105 +7,141 @@
...
         """
         Schedule the executions for each of the importers at their specified
         refresh time
         """
 
-        apis = self.get_apis()
+        apis = Scheduler.get_apis()
         for api in apis:
-            self.scheduler.add_job(self.fetch_data, 'interval',
+            sched.add_job(Scheduler.fetch_data, 'interval',
```
Comment (j0rdanl0urens) [inline Analytics/importers/scheduler_new.py]: Re-align

File: Analytics/importers/scheduler_new.py [hunk 4]
```diff
@@ -7,105 +7,141 @@
...
                                    seconds=api.refresh_time,
                                    start_date=datetime.now()+timedelta(seconds=5),
                                    end_date=datetime.now()+timedelta(hours=23),
-                                   args=[api.api_class,api.name])
-
-    def run(self):
+                                   args=[api.api_class, api.name],
+                                   replace_existing=True, id='{}'.format(api.api_class),
+                                   jobstore='sqlalchemy')
+    @staticmethod
```
Comment (j0rdanl0urens) [inline Analytics/importers/scheduler_new.py]: New line between functions

File: Analytics/importers/scheduler_new.py [hunk 5]
```diff
@@ -7,105 +7,141 @@
...
+        apis = Scheduler.get_apis()
+        for api in apis:
+            if not ImporterStatuses.find_by_api_id(api.id):
+                class_name = api.api_class.split('.')[2]
+                new_entry = ImporterStatuses(api.id, class_name, 'pending',
+                                             '', '', datetime.now())
+                new_entry.save()
+                new_entry.commit()
+
+        sched.add_job(Scheduler.main_task, 'interval',
```
Comment (j0rdanl0urens) [inline Analytics/importers/scheduler_new.py]: Align

File: Analytics/models/importer_status.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    id = db.Column(db.Integer, primary_key=True)
+    api_id = db.Column(db.Integer, nullable=False)
+    import_class_name = db.Column(db.String(100), nullable=False)
+    state = db.Column(db.String(50), nullable=False)
+    reason = db.Column(db.Text)
+    trace = db.Column(db.Text)
+    timestamp = db.Column(db.DateTime)
+
+    def __init__(self, api_id: int, import_class_name: str, state: str,
+                 reason: str, trace: str,timestamp: datetime = datetime.now()):
```
Comment (j0rdanl0urens) [inline Analytics/models/importer_status.py]: Docstrings

File: Analytics/models/importer_status.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        Return the Importer Status entry that matches the api_id argument
+        :param name: name of importer which is contained in the api table
+        :return: the Importer Status entry that match the api_id argument
+        """
+
+        return cls.query.filter_by(import_class_name=name).first()
+
+    @classmethod
+    def remove_where(cls, time_limit: datetime) -> db.Model:
+        """ Delete all entries entered after time_limit argument """
```
Comment (j0rdanl0urens) [inline Analytics/models/importer_status.py]: Add missing fields

File: Analytics/test_importer_status.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    @staticmethod
+    def remove_importer_entries(limit: datetime):
+        """
+        Remove the entries that were persisted by the simulate_importer method
+        :param limit: the date and time limit after which entries should be
+                      deleted
+        """
+
+        ImporterStatuses.remove_where(limit)
+
```
Comment (j0rdanl0urens) [inline Analytics/test_importer_status.py]: Please remove the extra lines

File: Analytics/importers/scheduler_new.py (hunk 2 above)
Comment (kpennels) [reply Analytics/importers/scheduler_new.py]: The self parameter is required for the @status_tracker decorator to function correctly. If it is removed, the corresponding import class name will not be returned

**Pull request merged**